### PR TITLE
Add .npy File I/O

### PR DESCRIPTION
**Pull request by AtomicVar**
This PR adds these two functions to `file_io.h` to support NumPy `.npy` file I/O:

- `io::read_csv`
- `io::write_csv`

I also fixed compile flag markups in File IO's rst docs.

### CODE REVIEW

File: docs_input/api/io/write_npy.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+.. _write_npy_func:
+
+write_npy
+=========
+
+Write a NPY file from a tensor
```
Comment (cliffburdick) [inline docs_input/api/io/write_npy.rst:6]: `an`

**Review by cliffburdick — APPROVED** [review]

**Comment by cliffburdick** [comment]
Thanks again @AtomicVar

**Comment by AtomicVar** [comment]
I fixed that typo. Could you check it again? @cliffburdick

**Pull request merged**